### PR TITLE
Feature/added-button-variant-hamburger

### PR DESCRIPTION
**Pull request by Shivam-apna**
please review and approve it.

### CODE REVIEW

File: application/backend/src/main/resources/apps/typerefinery/components/forms/fields/button/clientlibs/functions.js [hunk 1]
```diff
@@ -25,6 +25,28 @@ window.Typerefinery.Page.Theme = Typerefinery.Page.Theme || {};
             }
         });
     }
+ 
+    $(function () {
```
Comment (Harikarthyk) [inline application/backend/src/main/resources/apps/typerefinery/components/forms/fields/button/clientlibs/functions.js:29]: you can add it as a separate function in the ns.

File: application/backend/src/main/resources/apps/typerefinery/components/forms/fields/button/variant.hamburger.html [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+</template>
+<template data-sly-template.variant="${ @ model }">
+  <div class="${wcmmode.isEdit ? '':'navbar-expand-lg mt-2'}" data-model="${model.jsonString}">
+      <button component="${model.componentName}" id="${model.id}" type="${model.type}" data-bs-toggle="collapse"
+          data-bs-target="${model.target}" class="navbar-toggler">
+          <sly data-sly-call="${icon @ icon='pi pi-align-justify'}">
+          </sly>
+      </button>
+  </div>
+  <sly data-sly-test="${wcmmode.isEdit}">
```
Comment (Harikarthyk) [inline application/backend/src/main/resources/apps/typerefinery/components/forms/fields/button/variant.hamburger.html:28]: Please format this code.

File: tests/content/src/main/content/jcr_root/content/typerefinery-showcase/pages/content/footer/.content.xml [hunk 3]
```diff
@@ -33,10 +33,10 @@
...
           <title jcr:primaryType="nt:unstructured" sling:resourceType="typerefinery/components/content/title" title ="Copyright © 2023 Typerefinery. /All Rights Reserved. /Privacy Terms of Use" headingLevel="h6"/>
         </container1>
-        <container2 jcr:primaryType="nt:unstructured" sling:resourceType="typerefinery/components/layout/container" lgColSize="4" mdColSize="12" smColSize="12" gap="3" flexEnabled="true" horizontalAlignment="end">
-          <button_1 jcr:primaryType="nt:unstructured" sling:resourceType="typerefinery/components/forms/fields/button" buttonType="navigate" buttonVariant="primary" isOutlinedButton="true" label="Github" showIcon="true" icon="pi pi-github" lgColSize="4" mdColSize="6" navigateTo="https://github.com/typerefinery-ai/typerefinery-websight" openInNewTab="true" classes="btn-sm"></button_1>
+        <container2 jcr:primaryType="nt:unstructured" sling:resourceType="typerefinery/components/layout/container" lgColSize="4" mdColSize="12" smColSize="12" columnGap="3" flexEnabled="true" horizontalAlignment="end">
```
Comment (Harikarthyk) [inline tests/content/src/main/content/jcr_root/content/typerefinery-showcase/pages/content/footer/.content.xml:36]: why are you updating gap to columnGap ?

File: tests/content/src/main/content/jcr_root/content/typerefinery-showcase/pages/content/header/.content.xml [hunk 4]
```diff
@@ -5,10 +5,10 @@
...
-      <container2 jcr:primaryType="nt:unstructured" sling:resourceType="typerefinery/components/layout/container" lgColSize="12" mdColSize="12" smColSize="12" gap="3" flexEnabled="true" horizontalAlignment="end">
-        <button_1 jcr:primaryType="nt:unstructured" sling:resourceType="typerefinery/components/forms/fields/button" label="Typerefinery.ai" buttonType="navigate" buttonVariant="warning" showIcon="true" icon="pi pi-file"  lgColSize="2" mdColSize="6" navigateTo="https://typerefinery.ai/" persistColorWhenThemeSwitches="true" openInNewTab="true" classes="btn-sm"></button_1>
-        <button_3 jcr:primaryType="nt:unstructured" sling:resourceType="typerefinery/components/forms/fields/button" buttonType="action" showIcon="true" icon="	pi pi-sun " hideButtonLabel="true" actionType="initialTheme" toggleTheme="light"  lgColSize="1" mdColSize="6"  classes="btn-sm " persistColorWhenThemeSwitches="true" buttonVariant="primary"  ></button_3>
+      <container2 jcr:primaryType="nt:unstructured" sling:resourceType="typerefinery/components/layout/container" lgColSize="12" mdColSize="12" smColSize="12" columnGap="3" flexEnabled="true" horizontalAlignment="end">
```
Comment (Harikarthyk) [inline tests/content/src/main/content/jcr_root/content/typerefinery-showcase/pages/content/header/.content.xml:8]: same here

**Review by Harikarthyk — CHANGES_REQUESTED** [review]

**Comment by wildone** [comment]
@Harikarthyk is this good to go?

**Pull request merged**